File: libs/power-log-parser/src/lib/models/sub-spell.ts
export class SubSpell {
	Timestamp: string = '';
	Prefab: string = '';
	Source: number = 0;
	Targets: number[] = [];

	private _spell: SubSpell | null = null;
	private _parent: SubSpell | null = null;

	get Spell(): SubSpell | null {
		return this._spell;
	}

	set Spell(value: SubSpell | null) {
		this._spell = value;
		if (this._spell != null) {
			this._spell._parent = this;
		}
	}

	get Parent(): SubSpell | null {
		return this._parent;
	}

	GetActiveSubSpell(): SubSpell {
		let current: SubSpell = this;
		while (current?.Spell != null) {
			current = current.Spell;
		}
		return current;
	}
}

File: tools/linux-helpers/hs-converter/HearthstoneReplays/Parser/SubSpell.cs
#region

using System;
using System.Collections.Generic;

#endregion

namespace HearthstoneReplays.Parser
{
	public class SubSpell
	{
		public DateTime Timestamp { get; set; }
		public string Prefab { get; set; }
		public int Source { get; set; }
		public IList<int> Targets { get; set; }

		public SubSpell Spell { 
			get {
				return _subSpell;
			}
			set
			{
				this._subSpell = value;
				if (this.Spell != null) { 
					this._subSpell.Parent = this;
				}
			}
		}
		private SubSpell _subSpell;

		public SubSpell Parent { get; private set; }

        public SubSpell GetActiveSubSpell()
        {
			var current = this;
			while (current?.Spell != null)
			{
				current = current.Spell;
			}
			return current;
        }
    }
}